pkg/deployment/cert.go

```diff
@@ -175,20 +174,17 @@ func EnsureTLSCerts(ctx context.Context, helper *helper.Helper,
 		// NOTE: we are assuming that there will always be a ctlplane network
 		// that means if you are not using network isolation with multiple networks
 		// you should still need to have a ctlplane network at a minimum to use tls-e
-		basename := allHostnames[nodeName][CtlPlaneNetwork]
-		// in case the control plane network is not present we will fall back to the
-		// hostname, and log a warning.
-		field := reflect.ValueOf(basename)
-		if field.IsZero() {
-			basename = hostName
-			helper.GetLogger().Error(fmt.Errorf(
-				"control plane network not found for node %s, falling back to hostname", nodeName),
-				"tls-e requires a control plane network to be present")
+
+		baseName, ok := dnsNames[CtlPlaneNetwork]
+		if !ok {
+			return nil, fmt.Errorf(
```

Comment:
should that return &result, ...

Reply:
Would not that be nil?

Reply:
I can change it to align with the other parts of the code, but it would surely be nil here as we return the only place it could have something else.